app/controllers/password_resets_controller.rb

```diff
@@ -0,0 +1,48 @@
+class PasswordResetsController < ApplicationController
+  require_unauthenticated_access
+  before_action :ensure_manual_login_allowed
+
+  layout "public"
+
+  def new
+  end
+
+  def create
+    os = request.operating_system
+    browser = request.browser
+    SendResetPasswordEmailJob.perform_later(params[:email], os, browser) # queue as a job to avoid timing attacks
+
+    redirect_to '/login', notice: 'If an account with that email was found, we have sent a link to reset the password'
+  end
+
+  def edit
+    @person = find_signed_person(params[:token])
+  end
+
+  def update
+    @person = find_signed_person(params[:token])
+    password_credential = @person&.personable&.password_credential
```

Comment:
Let's use .user for readability, plus we wouldn't want to update a password for a Tombstone or other odd state of a user. Also, that first @person is guaranteed so only need the second &, right? Actually, as I look at that rescue I'm not sure why we only rescue on Invalid. Let me comment below.

```suggestion
    password_credential = @person.user&.password_credential
```